### PR TITLE
Allow creating ConstantScalarNode for double type

### PR DESCRIPTION
**Pull request by pengwa**
### Allow creating ConstantScalarNode for double type

Allow create ConstantScalarNode for double type. Looks double type is not respected when creating constant. So fix it.

```
onnxruntime::python::addObjectMethodsForTraining(pybind11::module&, onnxruntime::python::ExecutionProviderRegistrationFn)::<lambda(onnxruntime::training::OrtModuleGraphBuilder*, const onnxruntime::training::TrainingGraphTransformerConfiguration&)> [ONNXRuntimeError] : 1 : FAIL : Type Error: Type parameter (T) of Optype (Sub) bound to different types (tensor(double) and tensor(float) in node (/_original_module/_original_model/gpt_neox/layers.0/input_layernorm/Pow_Grad/Sub_1).
```

### CODE REVIEW

File: orttraining/orttraining/core/framework/gradient_graph_builder.cc [hunk 1]
```diff
@@ -123,8 +123,9 @@ NodeSet GradientGraphBuilder::BFSWithStopGradient(const std::unordered_set<std::
     std::vector<const Node*> nodes = graph_->GetConsumerNodes(name);
     for (const Node* node : nodes) {
       int input_index = graph_utils::GetNodeInputIndexFromInputName(*node, name);
-      const std::unordered_set<size_t>* edges = GetStopGradientEdges(*node);
-      if (edges != nullptr && edges->count(input_index)) {
+      std::unordered_set<size_t> edges;
```
Comment (askhade) [inline orttraining/orttraining/core/framework/gradient_graph_builder.cc:126]: I don't understand the reason behind this change. Can you please explain?

Comment (pengwa) [reply orttraining/orttraining/core/framework/gradient_graph_builder.cc:126]: Ok, let me add more details in the PR description. :)

**Comment by pengwa** [comment]
Need check again whether this is still needed or not.

File: orttraining/orttraining/core/framework/gradient_graph_builder.cc (hunk 1 above)
Comment (pengwa) [reply orttraining/orttraining/core/framework/gradient_graph_builder.cc]: removed gradient build stuff, only fix a constant creation bug. Will create a new one if gradient builder still need the fix.

**Review by ajindal1 — APPROVED** [review]

**Comment by pengwa** [comment]
Merged this small fix to unblock gradient build for stage 3. Let me know if you have more comment @askhade  .

**Pull request merged**